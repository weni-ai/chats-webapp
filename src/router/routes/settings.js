const routes = [
  {
    path: '/settings',
    component: () => import('@/layouts/SettingsLayout/index.vue'),
    children: [
      {
        path: '/settings',
        name: 'sectors',
        component: () => import('@/views/Settings/index.vue'),
      },

      {
        path: 'sectors/new',
        name: 'sectors.new',
        component: () => import('@/views/Settings/Sectors/New.vue'),
      },
      {
        path: 'sectors/:uuid',
        name: 'sectors.edit',
        component: () => import('@/views/Settings/Sectors/Edit.vue'),
        props: (route) => ({
          uuid: route.params.uuid,
          tab: route.query.tab,
        }),
      },
    ],
  },
];

export default routes;
