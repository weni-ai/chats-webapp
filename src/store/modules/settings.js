import { defineStore } from 'pinia';
import Sector from '@/services/api/resources/settings/sector';
import cloneDeep from 'lodash.clonedeep';
import { removeDuplicatedItems } from '@/utils/array';

export const useSettings = defineStore('settings', {
  state: () => ({
    sectors: [],
    isLoadingSectors: false,
    nextSectors: '',
    previousSectors: '',
    currentSector: null,
  }),

  actions: {
    async getSectors() {
      const isInLastPage = !this.nextSectors && this.previousSectors;
      if (this.isLoadingSectors || isInLastPage) {
        return;
      }

      try {
        this.isLoadingSectors = true;
        const { results, next, previous } = await Sector.list({
          nextReq: this.nextSectors,
        });
        this.sectors = removeDuplicatedItems([...this.sectors, ...results]);
        this.nextSectors = next;
        this.previousSectors = previous;
      } catch (error) {
        console.error(error);
      } finally {
        this.isLoadingSectors = false;
      }
    },

    addSector(sector) {
      const lastId = this.sectors.at(-1).id;
      let queueId = 100;
      this.sectors.push({
        ...sector,
        id: lastId + 1,
        queues: sector.queues.map((q) => ({ ...q, id: queueId++ })),
      });
    },

    async getCurrentSector(uuid) {
      this.currentSector = await Sector.find(uuid);
    },

    saveSector(sector) {
      this.addSector({ ...sector, contacts: { count: 0 } });
    },

    updateSector(sector) {
      if (!sector.id) this.saveSector(sector);
      const index = this.sectors.findIndex((s) => s.id === sector.id);
      this.sectors.splice(index, 1, sector);
    },

    async deleteSector(sectorUuid) {
      await Sector.deleteSector(sectorUuid);
      this.sectors = this.sectors.filter(
        (sector) => sector.uuid !== sectorUuid,
      );
    },
  },

  getters: {
    getSectorById({ sectors }) {
      return (id) => {
        const sector = sectors.find((sector) => sector.id === id);
        return sector ? cloneDeep(sector) : null;
      };
    },
  },
});
