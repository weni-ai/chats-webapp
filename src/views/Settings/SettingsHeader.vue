<template>
  <header class="settings-view__header">
    <UnnnicAvatarIcon
      icon="settings"
      scheme="weni-600"
    />
    <h1 class="header__title">{{ $t('config_chats.title') }}</h1>
  </header>
</template>
<script>
export default {
  name: 'SettingsHeader',
};
</script>

<style lang="scss" scoped>
.settings-view__header {
  display: flex;
  align-items: center;
  gap: $unnnic-spacing-xs;

  .header__title {
    color: $unnnic-color-neutral-darkest;
    font-family: $unnnic-font-family-primary;
    font-size: $unnnic-font-size-title-sm;
    font-weight: $unnnic-font-weight-bold;
    line-height: $unnnic-font-size-title-sm + $unnnic-line-height-md;
  }
}
</style>
