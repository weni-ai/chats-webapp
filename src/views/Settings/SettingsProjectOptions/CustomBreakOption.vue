<template>
  <UnnnicButton
    class="custom-btn"
    :text="$t('config_chats.custom_breaks.title')"
    type="secondary"
    size="large"
    @click.prevent.stop="openModal"
  />

  <UnnnicModalDialog
    :modelValue="showModal"
    :title="$t('config_chats.custom_breaks.title')"
    :primaryButtonProps="{
      text: $t('save'),
      loading: isLoadingCloseRoom,
      disabled: !canSave,
    }"
    :secondaryButtonProps="{ text: $t('cancel') }"
    size="md"
    data-testid="chat-classifier-modal"
    showCloseIcon
    class="custom-breaks-modal"
    @primary-button-click="saveStatus"
    @secondary-button-click="closeModal"
    @update:model-value="closeModal"
  >
    <section class="custom-breaks-modal__content">
      <header>
        <p class="custom-breaks-modal__description">
          {{ $t('config_chats.custom_breaks.description') }}
        </p>
      </header>

      <section class="custom-breaks-modal__form">
        <UnnnicInput
          v-model="customBreakName"
          :label="$t('config_chats.custom_breaks.fields.name')"
          :placeholder="$t('config_chats.custom_breaks.fields.placeholder')"
          data-testid="new-status-name"
          class="custom-breaks-modal__input"
          :disabled="isLimitReached"
          @input="validateInput"
          @keyup.enter="addStatus"
        />
        <UnnnicButton
          :text="$t('add')"
          type="secondary"
          :disabled="isLimitReached || !customBreakName.trim() || isDuplicate"
          @click="addStatus"
        />
      </section>

      <section
        v-if="errorMessage"
        class="custom-breaks-modal__error-content"
      >
        <UnnnicIcon
          icon="alert-circle-1"
          :next="true"
          filled
          size="sm"
          scheme="feedback-red"
        />
        <p class="custom-breaks-modal__error-text">
          {{ errorMessage }}
        </p>
      </section>

      <section class="custom-breaks-modal__list">
        <div
          v-for="(status, index) in customBreaks"
          :key="index"
          class="custom-breaks-modal__item"
        >
          <p class="custom-breaks-modal__list__status">{{ status }}</p>
          <UnnnicButtonIcon
            icon="close"
            size="sm"
            class="delete-icon"
            type="tertiary"
            @click.prevent.stop="removeStatus(index)"
          />
        </div>
      </section>
    </section>
  </UnnnicModalDialog>
</template>

<script setup>
import { ref, computed } from 'vue';
import i18n from '@/plugins/i18n';
import callUnnnicAlert from '@/utils/callUnnnicAlert';

const showModal = ref(false);
const isLoadingCloseRoom = ref(false);
const customBreakName = ref('');
const customBreaks = ref([]);
const originalBreaks = ref([]);
const isDuplicate = ref(false);

const MAX_STATUS = 6;

const fetchCustomBreaks = async () => {
  return new Promise((resolve) => {
    setTimeout(() => {
      const fakeData = ['Em pausa', 'Em reuniÃ£o'];
      originalBreaks.value = [...fakeData];
      customBreaks.value = [...fakeData];
      resolve(fakeData);
    }, 500);
  });
};

const isLimitReached = computed(() => customBreaks.value.length >= MAX_STATUS);

const canSave = computed(() => {
  const hasNewStatus = customBreaks.value.some(
    (status) => !originalBreaks.value.includes(status),
  );
  return customBreaks.value.length > 0 && hasNewStatus;
});

const errorMessage = computed(() => {
  if (isLimitReached.value)
    return i18n.global.t('config_chats.custom_breaks.status_exceeded_limit');
  if (isDuplicate.value)
    return i18n.global.t('config_chats.custom_breaks.status_duplicate_name');
  return '';
});

const validateInput = () => {
  const lowerCaseName = customBreakName.value.trim().toLowerCase();
  isDuplicate.value = customBreaks.value.some(
    (status) => status.toLowerCase() === lowerCaseName,
  );
};

const addStatus = () => {
  if (
    !isLimitReached.value &&
    customBreakName.value.trim() &&
    !isDuplicate.value
  ) {
    customBreaks.value.push(customBreakName.value.trim());
    customBreakName.value = '';
    isDuplicate.value = false;
  }
};

const removeStatus = (index) => {
  customBreaks.value.splice(index, 1);
};

const saveStatus = async () => {
  if (!canSave.value) return;

  isLoadingCloseRoom.value = true;
  setTimeout(() => {
    originalBreaks.value = [...customBreaks.value];
    isLoadingCloseRoom.value = false;
    callUnnnicAlert({
      props: {
        text: i18n.global.t('config_chats.custom_breaks.status_success'),
        type: 'success',
      },
      seconds: 5,
    });
    closeModal();
  }, 1000);
};

const openModal = async () => {
  showModal.value = true;
  await fetchCustomBreaks();
};

const closeModal = () => {
  showModal.value = false;
};
</script>

<style lang="scss" scoped>
.custom-btn {
  width: 100%;
}
.custom-breaks-modal {
  &__description {
    color: $unnnic-color-neutral-cloudy;
    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-gt;
    font-style: normal;
    font-weight: $unnnic-font-weight-regular;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
    padding-bottom: $unnnic-spacing-sm;
  }

  &__form {
    display: flex;
    width: 100%;
    align-items: end;
    gap: $unnnic-spacing-sm;
  }

  &__input {
    width: 100%;
  }

  &__error-content {
    display: flex;
    padding: $unnnic-spacing-sm;
    align-items: center;
    gap: $unnnic-spacing-xs;
    align-self: stretch;
    border-radius: $unnnic-spacing-nano;
    border: 1px solid $unnnic-color-neutral-soft;
    background: $unnnic-color-background-carpet;
    margin-top: $unnnic-spacing-xs;
  }

  &__error-text {
    color: $unnnic-color-neutral-dark;

    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-gt;
    font-style: normal;
    font-weight: $unnnic-font-weight-regular;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  }

  &__list {
    margin-top: $unnnic-spacing-sm;
    display: flex;
    flex-direction: column;
    gap: $unnnic-spacing-nano;

    &__status {
      color: $unnnic-color-neutral-dark;

      font-family: $unnnic-font-family-secondary;
      font-size: $unnnic-font-size-body-gt;
      font-style: normal;
      font-weight: $unnnic-font-weight-bold;
      line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
    }
  }

  &__item {
    display: flex;
    justify-content: space-between;
    align-items: center;
    align-self: stretch;
  }
}
.delete-icon {
  cursor: pointer;
  background-color: inherit;
}
</style>
