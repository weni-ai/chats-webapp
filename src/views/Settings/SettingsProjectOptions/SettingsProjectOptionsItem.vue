<template>
  <section class="project-options__config">
    <UnnnicSwitch
      :modelValue="modelValue"
      :textRight="name"
      @update:model-value="$emit('update:model-value', $event)"
    />
    <UnnnicToolTip
      v-if="tooltip"
      enabled
      :text="tooltip"
      side="right"
      maxWidth="20rem"
    >
      <UnnnicIcon
        icon="info"
        scheme="neutral-soft"
        filled
        size="sm"
      />
    </UnnnicToolTip>
  </section>
</template>
<script>
export default {
  name: 'SettingsProjectOptionsItem',

  props: {
    modelValue: {
      type: Boolean,
      default: false,
    },
    name: {
      type: String,
      required: true,
    },
    tooltip: {
      type: String,
      default: '',
    },
  },

  emits: ['update:model-value'],
};
</script>
<style lang=""></style>
