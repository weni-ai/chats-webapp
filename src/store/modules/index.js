import auth from './auth';
import chats from './chats';
import settings from './settings';
import rooms from './rooms';

export default {
  auth,
  chats,
  settings,
  rooms,
};
