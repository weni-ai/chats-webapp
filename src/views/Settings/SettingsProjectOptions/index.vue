<template>
  <section
    v-if="isUserManager && projectConfig"
    class="settings-view__project-options"
  >
    <SettingsSectionHeader :title="$t('config_chats.project_configs.title')" />

    <section class="project-options__items">
      <section>
        <SettingsProjectOptionsItem
          v-model="projectConfig.can_use_bulk_transfer"
          :name="configBulkTransferTranslation"
          :tooltip="$t('config_chats.project_configs.bulk_transfer.tooltip')"
        />
        <SettingsProjectOptionsItem
          v-model="projectConfig.can_use_queue_prioritization"
          :name="configQueuePrioritizationTranslation"
          :tooltip="
            $t('config_chats.project_configs.queue_prioritization.tooltip')
          "
        />
        <SettingsProjectOptionsItem
          v-model="projectConfig.filter_offline_agents"
          :name="configBlockTransferToOffAgentsTranslation"
        />
        <SettingsProjectOptionsItem
          v-model="projectConfig.can_see_timer"
          :name="configShowAgentStatusCountTimer"
        />
      </section>
      <section class="project-options__items__custom-breaks">
        <CustomBreakOption />
      </section>
    </section>
  </section>
</template>

<script>
import { mapState } from 'pinia';

import { useConfig } from '@/store/modules/config';
import { useProfile } from '@/store/modules/profile';

import Project from '@/services/api/resources/settings/project';

import SettingsProjectOptionsItem from './SettingsProjectOptionsItem.vue';
import SettingsSectionHeader from '../SettingsSectionHeader.vue';
import CustomBreakOption from './CustomBreakOption.vue';

export default {
  name: 'SettingsProjectOptions',

  components: {
    SettingsSectionHeader,
    SettingsProjectOptionsItem,
    CustomBreakOption,
  },

  data() {
    return {
      projectConfig: {
        can_use_bulk_transfer: false,
        can_use_queue_prioritization: false,
        filter_offline_agents: false,
        can_see_timer: false,
      },
    };
  },

  computed: {
    ...mapState(useConfig, ['project']),
    ...mapState(useProfile, ['me']),

    isUserManager() {
      const ROLE_MANAGER = 1;
      return this.me.project_permission_role === ROLE_MANAGER;
    },

    configBulkTransferTranslation() {
      const canBulkTransfer = this.projectConfig.can_use_bulk_transfer;
      return this.$t(
        `config_chats.project_configs.bulk_transfer.switch_${
          canBulkTransfer ? 'active' : 'inactive'
        }`,
      );
    },
    configQueuePrioritizationTranslation() {
      const canQueuePrioritization =
        this.projectConfig.can_use_queue_prioritization;
      return this.$t(
        `config_chats.project_configs.queue_prioritization.switch_${
          canQueuePrioritization ? 'active' : 'inactive'
        }`,
      );
    },
    configBlockTransferToOffAgentsTranslation() {
      const canBulkTransfer = this.projectConfig.filter_offline_agents;
      return this.$t(
        `config_chats.project_configs.block_transfer_to_off_agents.switch_${
          canBulkTransfer ? 'active' : 'inactive'
        }`,
      );
    },
    configShowAgentStatusCountTimer() {
      const showAgentStatusCountTimer = this.projectConfig.can_see_timer;
      return this.$t(
        `config_chats.project_configs.show_agent_status_count_timer.switch_${
          showAgentStatusCountTimer ? 'active' : 'inactive'
        }`,
      );
    },
  },

  watch: {
    project: {
      immediate: true,
      handler(newProject) {
        if (newProject.config) {
          this.projectConfig = newProject.config;
        }
      },
    },
    projectConfig: {
      deep: true,
      handler() {
        this.updateProjectConfig();
      },
    },
  },

  methods: {
    async updateProjectConfig() {
      const {
        can_use_bulk_transfer,
        can_use_queue_prioritization,
        filter_offline_agents,
        can_see_timer,
      } = this.projectConfig;

      Project.update({
        can_use_bulk_transfer,
        can_use_queue_prioritization,
        filter_offline_agents,
        can_see_timer,
      });
    },
  },
};
</script>

<style lang="scss">
.settings-view__project-options {
  display: grid;
  gap: $unnnic-spacing-ant;

  .project-options__items {
    display: flex;
    justify-content: space-between;
    gap: $unnnic-spacing-nano;

    &__custom-breaks {
      display: flex;
      min-width: 236px;
    }
  }
}
</style>
