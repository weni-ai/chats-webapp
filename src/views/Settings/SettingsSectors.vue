<template>
  <SettingsSectionHeader
    data-testid="settings-sectors-header"
    :title="$t('config_chats.section_sectors_title', { project: projectName })"
    :subtitle="$t('config_chats.section_sectors_subtitle')"
  />
  <section class="settings-sectors__filters">
    <UnnnicInput
      v-model="sectorNameFilter"
      iconLeft="search-1"
      size="md"
      :placeholder="$t('search')"
    />
    <ListOrdinator
      v-model="sectorOrder"
      :label="$t('order_by.label')"
    />
  </section>

  <section class="settings-view__sectors">
    <UnnnicCard
      data-testid="settings-sectors-blank-card"
      type="blank"
      :text="$t('config_chats.new_sector')"
      icon="add"
      class="sectors__new-sector-card"
      @click="openNewSectorModal()"
    />

    <UnnnicSimpleCard
      v-for="sector in sectorsOrdered"
      :key="sector.id"
      :title="sector.name"
      clickable
      class="sectors__card"
      data-testid="settings-sectors-sector-card"
      @click="navigate('sectors.edit', { uuid: sector.uuid })"
    >
      <template #headerSlot>
        <UnnnicDropdown position="top-left">
          <template #trigger>
            <UnnnicToolTip
              enabled
              :text="$t('quick_messages.delete_or_edit')"
              side="left"
            >
              <UnnnicButton
                iconCenter="more_vert"
                type="tertiary"
                data-testid="open-dropdown-menu-button"
              />
            </UnnnicToolTip>
          </template>
          <UnnnicDropdownItem
            data-testid="dropdown-edit"
            @click="navigate('sectors.edit', { uuid: sector.uuid })"
          >
            <section class="dropdown-item-content">
              <UnnnicIconSvg
                class="icon"
                icon="edit_square"
                size="sm"
              />
              <p>{{ $t('edit') }}</p>
            </section>
          </UnnnicDropdownItem>
          <UnnnicDropdownItem
            data-testid="dropdown-delete"
            @click.stop="handlerOpenDeleteSectorModal(sector)"
          >
            <section
              class="dropdown-item-content dropdown-item-content__delete"
            >
              <UnnnicIconSvg
                class="icon"
                icon="delete"
                size="sm"
                scheme="danger"
              />
              <p>{{ $t('exclude') }}</p>
            </section>
          </UnnnicDropdownItem>
        </UnnnicDropdown>
      </template>
    </UnnnicSimpleCard>
  </section>

  <section
    v-if="isLoadingSectors"
    data-testid="settings-sectors-loading-section"
    class="settings-view__loading-sectors"
  >
    <img
      src="@/assets/LogoWeniAnimada4.svg"
      width="40"
    />
  </section>
  <NewSectorDrawer
    v-if="showNewSectorModal"
    v-model="showNewSectorModal"
    data-testid="new-sector-drawer"
    @close="closeNewSectorModal()"
  />
  <UnnnicModalNext
    v-if="showDeleteSectorModal"
    type="alert"
    icon="alert-circle-1"
    scheme="feedback-red"
    :title="$t('delete_sector') + ` ${toDeleteSector.name}`"
    :description="$t('cant_revert')"
    :validate="`${toDeleteSector.name}`"
    :validatePlaceholder="`${toDeleteSector.name}`"
    :validateLabel="
      $t('confirm_typing') + ` &quot;${toDeleteSector.name}&quot;`
    "
    :actionPrimaryLabel="$t('confirm')"
    :actionSecondaryLabel="$t('cancel')"
    data-testid="modal-delete-sector"
    @click-action-primary="deleteSector(toDeleteSector.uuid)"
    @click-action-secondary="handlerCloseDeleteSectorModal()"
  />
</template>

<script>
import unnnic from '@weni/unnnic-system';
import { mapActions, mapState } from 'pinia';

import { useConfig } from '@/store/modules/config';
import { useSettings } from '@/store/modules/settings';

import SettingsSectionHeader from './SettingsSectionHeader.vue';

import NewSectorDrawer from './Sectors/New/NewSectorDrawer.vue';
import ListOrdinator from '@/components/settings/ListOrdinator.vue';

export default {
  name: 'SettingsSectors',

  components: {
    SettingsSectionHeader,
    NewSectorDrawer,
    ListOrdinator,
  },

  data() {
    return {
      showNewSectorModal: false,
      showDeleteSectorModal: false,
      sectorNameFilter: '',
      sectorOrder: 'alphabetical',
      toDeleteSector: {},
    };
  },

  computed: {
    ...mapState(useConfig, {
      projectName: (store) => store.project.name,
    }),
    ...mapState(useSettings, ['sectors', 'isLoadingSectors']),

    sectorsOrdered() {
      let sectorsOrdered = this.sectors.slice().sort((a, b) => {
        let first = null;
        let second = null;

        if (this.sectorOrder === 'alphabetical') {
          first = a.name.toLowerCase();
          second = b.name.toLowerCase();
        } else if (this.sectorOrder === 'newer') {
          first = new Date(b.created_on).getTime();
          second = new Date(a.created_on).getTime();
        } else if (this.sectorOrder === 'older') {
          first = new Date(a.created_on).getTime();
          second = new Date(b.created_on).getTime();
        }

        return first === second ? 0 : first > second ? 1 : -1;
      });

      return this.sectorNameFilter.trim()
        ? sectorsOrdered.filter(({ name }) =>
            name
              .toLowerCase()
              .includes(this.sectorNameFilter.trim().toLowerCase()),
          )
        : sectorsOrdered;
    },
  },

  methods: {
    ...mapActions(useSettings, {
      actionDeleteSector: 'deleteSector',
    }),
    handlerOpenDeleteSectorModal(sector) {
      this.toDeleteSector = sector;
      this.handleConnectOverlay(true);
      this.showDeleteSectorModal = true;
    },

    handlerCloseDeleteSectorModal() {
      this.toDeleteSector = {};
      this.handleConnectOverlay(false);
      this.showDeleteSectorModal = false;
    },

    async deleteSector(sectorUuid) {
      try {
        await this.actionDeleteSector(sectorUuid);

        this.$router.push({ name: 'sectors' });
        unnnic.unnnicCallAlert({
          props: {
            text: this.$t('sector_deleted_success'),
            type: 'success',
          },
          seconds: 5,
        });
      } catch (error) {
        console.log(error);
        unnnic.unnnicCallAlert({
          props: {
            text: this.$t('sector_delete_error'),
            type: 'error',
          },
          seconds: 5,
        });
      } finally {
        this.showDeleteSectorModal = false;
        this.handleConnectOverlay(false);
      }
    },
    handleConnectOverlay(active) {
      window.parent.postMessage({ event: 'changeOverlay', data: active }, '*');
    },
    openNewSectorModal() {
      setTimeout(() => {
        this.handleConnectOverlay(true);
      }, 1);
      this.showNewSectorModal = true;
    },
    closeNewSectorModal() {
      setTimeout(() => {
        this.handleConnectOverlay(false);
      }, 1);
      this.showNewSectorModal = false;
    },
    navigate(name, params) {
      this.$router.push({
        name,
        params,
      });
    },
  },
};
</script>

<style lang="scss" scoped>
.settings-sectors {
  &__filters {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    gap: $unnnic-spacing-stack-sm $unnnic-spacing-inline-md;

    .unnnic-form {
      flex: 1;
    }
  }
}
.settings-view {
  &__sectors {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    gap: $unnnic-spacing-sm;

    .sectors__card {
      min-height: 120px;

      :deep(.unnnic-simple-card-header-container__title) {
        color: $unnnic-color-neutral-darkest;
      }
    }
  }

  .sectors__new-sector-card:hover {
    box-shadow: $unnnic-shadow-level-far;
  }
  .sectors__new-sector-card:active {
    border: 1px solid $unnnic-color-neutral-cleanest;
  }
  .sectors__new-sector-card {
    min-height: 120px;
    :deep(.unnnic-card-blank__content) {
      flex-direction: row;
    }
    :deep(.unnnic-card-blank__content__icon) {
      font-size: $unnnic-font-size-title-md;
    }
  }

  .dropdown-item-content {
    display: flex;
    align-items: center;
    gap: $unnnic-spacing-xs;

    white-space: nowrap;

    &__delete {
      color: red;
    }
  }

  &__loading-sectors {
    display: flex;
    align-items: center;
    justify-content: center;
    padding: $unnnic-spacing-sm;
  }
}
</style>
