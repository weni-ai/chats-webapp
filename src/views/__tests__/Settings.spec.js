import { mount } from '@vue/test-utils';
import { describe, it, expect, vi, beforeEach } from 'vitest';

import { createTestingPinia } from '@pinia/testing';

import { useSettings } from '@/store/modules/settings';

import SettingView from '@/views/Settings/index.vue';
import SettingsHeader from '@/views/Settings/SettingsHeader.vue';
import SettingsProjectOptions from '@/views/Settings/SettingsProjectOptions/index.vue';
import SettingsSectors from '@/views/Settings/SettingsSectors.vue';

const createWrapper = (props = {}) => {
  return mount(SettingView, {
    props,
    global: {
      plugins: [createTestingPinia()],
    },
  });
};

describe('SettingView.vue', () => {
  let wrapper;
  let store;
  let settingsView;

  const updateWrapperProperty = (property, value) =>
    Object.defineProperty(wrapper.element, property, {
      value,
      writable: true,
    });

  beforeEach(() => {
    wrapper = createWrapper();
    store = useSettings();
    settingsView = wrapper.find('[data-testid=settings-view]');

    updateWrapperProperty('scrollTop', 0);
    updateWrapperProperty('clientHeight', 500);
    updateWrapperProperty('scrollHeight', 1000);
  });

  it('should render SettingsHeader, SettingsProjectOptions, and SettingsSectors components', () => {
    expect(wrapper.findComponent(SettingsHeader).exists()).toBe(true);
    expect(wrapper.findComponent(SettingsProjectOptions).exists()).toBe(true);
    expect(wrapper.findComponent(SettingsSectors).exists()).toBe(true);
  });

  it('should call getSectors on beforeMount', () => {
    expect(store.getSectors).toHaveBeenCalled();
  });

  it('should not call getSectors if not scrolling to the bottom', async () => {
    const getSectorsMock = vi.spyOn(store, 'getSectors');

    await settingsView.trigger('scroll');

    expect(getSectorsMock).toHaveBeenCalledTimes(0);
  });

  it('should match the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
