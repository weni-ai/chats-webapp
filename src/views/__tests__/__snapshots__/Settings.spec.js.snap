// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SettingView.vue > should match the snapshot 1`] = `
<section
  class="settings-view"
  data-testid="settings-view"
  data-v-002f92b9=""
>
  <header
    class="settings-view__header"
    data-v-002f92b9=""
    data-v-a59c663f=""
  >
    <div
      class="unnnic-avatar-icon weni-600 sm"
      data-testid="settings-view-header-icon"
      data-v-a59c663f=""
      data-v-d0742329=""
    >
      <span
        class="material-symbols-rounded unnnic-icon-scheme--weni-600 unnnic-icon-size--md"
        data-testid="material-icon"
        data-v-9846e6db=""
        data-v-d0742329=""
        translate="no"
      >
        settings
      </span>
    </div>
    <h1
      class="header__title"
      data-testid="settings-view-header-title"
      data-v-a59c663f=""
    >
      Chat Settings
    </h1>
  </header>
  
  <!--v-if-->
  
  <header
    class="settings-view__section-header"
    data-testid="settings-sectors-header"
    data-v-1088b44f=""
    data-v-f23b63f9=""
  >
    <p
      class="section-header__title"
      data-testid="section-header-title"
      data-v-f23b63f9=""
    >
      Sectors in 
    </p>
    <p
      class="section-header__subtitle"
      data-testid="section-header-subtitle"
      data-v-f23b63f9=""
    >
      Add, view and manage sectors, queues, managers and agents within your organization.
    </p>
  </header>
  <section
    class="settings-sectors__filters"
    data-v-1088b44f=""
  >
    <div
      class="unnnic-form md"
      data-v-1088b44f=""
      data-v-24f2d284=""
    >
      <!---->
      <div
        class="text-input size--md text-input--icon-right-size-sm unnnic-form-input"
        data-v-24f2d284=""
        data-v-c1401a33=""
        mask=""
      >
        <input
          class="unnnic-form-input input-itself input size-md normal input--has-icon-left unnnic-form-input input-itself"
          data-v-0caa2ccb=""
          data-v-c1401a33=""
          hascloudycolor="false"
          iconleft="search-1"
          iconrightsize="sm"
          placeholder="Search"
          type="text"
          value=""
        />
        <svg
          class="unnnic-icon unnnic-icon__size--sm unnnic-icon-scheme--neutral-cloudy icon-left"
          data-testid="old-map-icons"
          data-v-9846e6db=""
          data-v-c1401a33=""
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="primary"
            d="M34.0615 35.0005C33.8116 35.0005 33.5754 34.903 33.3992 34.7256L25.2575 26.5826C24.3065 27.3961 23.2355 28.0597 22.067 28.5559C20.5312 29.2082 18.9178 29.5394 17.272 29.5394C15.7011 29.5394 14.1578 29.2382 12.6819 28.6421C9.64389 27.4161 7.26574 25.0817 5.98481 22.0662C4.70388 19.052 4.67263 15.7178 5.89858 12.6811C7.12452 9.64433 9.45894 7.26492 12.4732 5.98399C14.009 5.33041 15.6211 5.00049 17.2682 5.00049C18.8391 5.00049 20.3825 5.30291 21.8583 5.89776C24.8963 7.12371 27.2745 9.45812 28.5554 12.4736C29.8363 15.4891 29.8676 18.822 28.6416 21.8588C28.1393 23.1035 27.4469 24.2444 26.5809 25.2554L34.7251 33.4009C35.09 33.7658 35.09 34.3606 34.7251 34.7268C34.5477 34.903 34.3115 35.0005 34.0615 35.0005ZM17.2682 6.87502C15.8748 6.87502 14.5077 7.15495 13.2068 7.70856C10.6524 8.79454 8.67413 10.809 7.63564 13.3821C6.59715 15.9552 6.6234 18.7783 7.70813 21.3327C8.7941 23.887 10.8086 25.8653 13.3817 26.9038C14.6326 27.4086 15.9411 27.6636 17.2707 27.6636C18.6641 27.6636 20.0313 27.3836 21.3322 26.83C22.5494 26.3127 23.6404 25.5828 24.5739 24.6606C24.5864 24.6431 24.6002 24.6281 24.6139 24.6143C24.6339 24.5943 24.6526 24.5781 24.6701 24.5656C25.6236 23.6021 26.381 22.4499 26.9021 21.1565C27.9406 18.5833 27.9143 15.7603 26.8283 13.2059C25.7436 10.6516 23.7279 8.67332 21.1548 7.63483C19.9063 7.13121 18.5979 6.87502 17.2682 6.87502Z"
            fill="#3B414D"
          />
        </svg>
        <!---->
      </div>
      <!---->
    </div>
    <div
      class="order"
      data-v-1088b44f=""
      data-v-348024a5=""
    >
      <p
        class="label"
        data-v-348024a5=""
      >
        Order by: 
      </p>
      <section
        class="ordinators"
        data-v-348024a5=""
      >
        
        <div
          class="unnnic-radio-container unnnic-radio-container--md"
          data-v-348024a5=""
          data-v-abd1b195=""
        >
          <svg
            class="unnnic-icon unnnic-icon__size--md unnnic-icon-scheme--brand-weni unnnic-radio"
            data-testid="old-map-icons"
            data-v-9846e6db=""
            data-v-abd1b195=""
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              class="primary"
              cx="20"
              cy="20.0005"
              fill="#3B414D"
              r="15"
            />
            <circle
              cx="20"
              cy="20.0005"
              fill="white"
              r="6.66667"
            />
          </svg>
          <span
            class="label"
            data-v-abd1b195=""
          >
            
            Alphabetical
            
          </span>
        </div>
        <div
          class="unnnic-radio-container unnnic-radio-container--md"
          data-v-348024a5=""
          data-v-abd1b195=""
        >
          <svg
            class="unnnic-icon unnnic-icon__size--md unnnic-icon-scheme--neutral-cleanest unnnic-radio"
            data-testid="old-map-icons"
            data-v-9846e6db=""
            data-v-abd1b195=""
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              class="primary-stroke"
              cx="20"
              cy="20.0005"
              r="14.5"
              stroke="#3B414D"
            />
          </svg>
          <span
            class="label"
            data-v-abd1b195=""
          >
            
            Newer
            
          </span>
        </div>
        <div
          class="unnnic-radio-container unnnic-radio-container--md"
          data-v-348024a5=""
          data-v-abd1b195=""
        >
          <svg
            class="unnnic-icon unnnic-icon__size--md unnnic-icon-scheme--neutral-cleanest unnnic-radio"
            data-testid="old-map-icons"
            data-v-9846e6db=""
            data-v-abd1b195=""
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              class="primary-stroke"
              cx="20"
              cy="20.0005"
              r="14.5"
              stroke="#3B414D"
            />
          </svg>
          <span
            class="label"
            data-v-abd1b195=""
          >
            
            Older
            
          </span>
        </div>
        
      </section>
    </div>
  </section>
  <section
    class="settings-view__sectors"
    data-v-1088b44f=""
  >
    <div
      buttonaction="() => {
      }"
      class="unnnic-card-blank unnnic-card-component sectors__new-sector-card sectors__new-sector-card"
      data-testid="settings-sectors-blank-card"
      data-v-1088b44f=""
      enabled="true"
      hasinformationicon="true"
      infoposition="right"
      percent="0"
      rating="0"
      scheme="aux-blue"
    >
      <div
        class="unnnic-card-blank__content"
      >
        <span
          class="material-symbols-rounded unnnic-icon-scheme--neutral-clean unnnic-icon-size--xl unnnic-card-blank__content__icon"
          data-testid="material-icon"
          data-v-9846e6db=""
          translate="no"
        >
          add
        </span>
        <div
          class="unnnic-card-blank__content__text"
        >
          New sector
        </div>
      </div>
    </div>
    
    
  </section>
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  
  
</section>
`;
